src/librustc_codegen_llvm/mir/operand.rs

```diff
@@ -145,6 +145,14 @@ impl OperandRef<'ll, 'tcx> {
         }
     }
 
+    /// Returns whether this operand refers to a scalar
+    pub fn is_immediate(self) -> bool {
```

Comment:
It doesn’t seem very necessary to add such a function when you can also just

```rust
if let OperandValue::Immediate(_) = val { 
    // ...
}
```

Reply:
Fixed in 33b67d9 !